File: java/java/src/main/java/org/signal/zkgroup/profiles/PniCredentialResponse.java
//
// Copyright 2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.profiles;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class PniCredentialResponse extends ByteArray {
  public PniCredentialResponse(byte[] contents) throws InvalidInputException {
    super(contents);
    try {
      Native.PniCredentialResponse_CheckValidContents(contents);
    } catch (IllegalArgumentException e) {
      throw new InvalidInputException(e.getMessage());
    }
  }
}

File: swift/Sources/SignalClient/zkgroup/PniCredentialResponse.swift
//
// Copyright 2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation
import SignalFfi

public class PniCredentialResponse: ByteArray {
  public required init(contents: [UInt8]) throws {
    try super.init(contents, checkValid: signal_pni_credential_response_check_valid_contents)
  }
}
